drivers/spi/spi_context.h

```diff
@@ -245,6 +245,25 @@ bool spi_context_rx_on(struct spi_context *ctx)
 	return !!(ctx->rx_buf || ctx->rx_len);
 }
 
+static ALWAYS_INLINE
+int spi_context_get_longest_single_xfer_buffers(struct spi_context *ctx,
+						void **tx_buf,
+						void **rx_buf)
+{
+	*tx_buf = ctx->tx_buf;
+	*rx_buf = ctx->rx_buf;
+
+	if (ctx->tx_len == 0) {
```

Comment:
OK, I see what this is doing now. Thanks for explaining. I see more why you have a use for this. Do you still want to update the argument pointers if both lengths are zero, then?